packages/app/src/components/nominationInfo/EditCard.js

```diff
@@ -0,0 +1,147 @@
+import React, { useState, useEffect } from 'react';
+import DatePicker from 'react-datepicker';
+import styles from './newstyles.module.css';
+import { Controller } from 'react-hook-form';
+import { Link } from 'react-router-dom';
+import './calendar.css';
+
+const EditCard = props => {
```

Comment:
Think there's any way to combine some of the logic between EditCard and ViewCard? There's a lot of overlap between these two.

Reply:
Yeah, components for the label types that are shared between the two would work. The goal isn't to reduce the number of times the switch statement runs, it's to reduce the amount of duplicated code.

Reply:
done